### PR TITLE
add confusion matrix

### LINKED ISSUE #744: [FEAT][CV] Confusion matrix for object detection
**Describe alternatives you've considered**
The confusion matrix check for object detection should be modified to include an IoU threshold, and the logic should be changed accordingly.

**Additional context**

### PR DESCRIPTION
**Pull request by benisraeldan**
#### Reference Issues/PRs

resolve #744 

#### What does this implement/fix? Explain your changes.
Add Confusion matrix report to vision.

### CODE REVIEW

File: deepchecks/vision/checks/performance/confusion_matrix.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            try:
+                detections_passed_threshold = [
+                    detection for detection in image_detections if detection[4] > self.confidence_threshold
+                ]
+            except IndexError:
+                # detections are empty, update matrix for labels
+                for label in image_labels:
+                    gt_class = label[0]
+                    self.matrix[self.num_classes, gt_class] += 1
+                continue
```
Comment (nirhutnik) [inline deepchecks/vision/checks/performance/confusion_matrix.py:147]: Not a programmer, but isn't try-except a relatively costly process? Wouldn't it be better to just check if detection[4] exists? also, doesn't this mean that you don't continue with creating the relevant list, thus skipping the detections after the empty one?

File: docs/source/examples/checks/vision/performance/confusion_martix.ipynb [hunk 2]
```diff
@@ -0,0 +1,9265 @@
+{
```
Comment (nirhutnik) [inline docs/source/examples/checks/vision/performance/confusion_martix.ipynb:1]: <p>There's supposed to be an available model in datasets/.../ mnist.py</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/828/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (nirhutnik) [inline docs/source/examples/checks/vision/performance/confusion_martix.ipynb:1]: <p>Can't see anything still :( Maybe the problem is in my browser, will check later</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/828/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (nirhutnik) [inline docs/source/examples/checks/vision/performance/confusion_martix.ipynb:1]: <p>Remove this writes to log</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/828/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

**Review by ItayGabbay — APPROVED** [review]

**Pull request merged**